airflow/providers/fab/auth_manager/security_manager/override.py

```diff
@@ -1093,7 +1094,7 @@ def is_dag_resource(self, resource_name: str) -> bool:
     def sync_perm_for_dag(
         self,
         dag_id: str,
-        access_control: dict[str, Collection[str]] | None = None,
```

Comment:
Instead of removing the type hint - make it up to date (same goes for the other mentions of `access_control` in methods' signatures)

Reply:
Thanks! Fixed in https://github.com/apache/airflow/pull/40703/commits/2e027f65df370234ad185230894db6a233196182